### PR TITLE
Adding a timeout value so users have a way to exit a search

### PR DESCRIPTION
**Pull request by monsur**

### CODE REVIEW

File: lib/sonos.js [hunk 1]
```diff
@@ -719,7 +719,7 @@ Sonos.prototype.getCurrentState = function(callback) {
  * Search "Class"
  * Emits 'DeviceAvailable' on a Sonos Component Discovery
  */
-var Search = function Search() {
+var Search = function Search(timeout) {
```
Comment (bencevans) [inline lib/sonos.js]: I'd rather this be an options object e.g `{ timeout: 10 * 1000 }` so we can add other arguments to the search functionality later without a crazy amount of arguments to the function.

File: lib/sonos.js [hunk 2]
```diff
@@ -758,10 +762,11 @@ util.inherits(Search, EventEmitter);
 /**
  * Create a Search Instance (emits 'DeviceAvailable' with a found Sonos Component)
  * @param  {Function} Optional 'DeviceAvailable' listener (sonos)
+ * @param  {int} Optional timeout value in milliseconds.
  * @return {Search/EventEmitter Instance}
  */
-var search = function(listener) {
-  var search = new Search();
+var search = function(listener, timeout) {
```
Comment (bencevans) [inline lib/sonos.js]: Options should come before the callback/listener function

**Comment by monsur** [comment]
I'll make that change. One question: how are you managing breaking changes? For this particular change, I can support both the form search(listener) and search(options, listener), but in general, do you discourage breaking changes, or do you recommend that breaking changes be made to a separate branch?

**Comment by bencevans** [comment]
If a change is breaking we keep it for the next major release usually :)
This change shouldn't need to be breaking so long as the arguments are
checked and if the first argument is a function, set options to an empty
object and the function as the callback

**Pull request closed without merging**